system-parachains/encointer/src/migrations_fix.rs

```diff
@@ -17,173 +17,114 @@
+				info!(target: TARGET, "initializing the set of invulnerables");
+
+				let raw_keys: Vec<[u8; 32]> = vec![
+					INVULNERABLE_A,
+					INVULNERABLE_B,
+					INVULNERABLE_C,
+					INVULNERABLE_D,
+					INVULNERABLE_E,
+				];
+
+				let validatorids: Vec<<T as pallet_session::Config>::ValidatorId> =
+					raw_keys.iter().map(|&pk| pk.into()).collect();
+
+				pallet_session::Validators::<T>::put(validatorids);
+
+				let queued_keys: Vec<(
+					<T as pallet_session::Config>::ValidatorId,
+					<T as pallet_session::Config>::Keys,
+				)> = raw_keys
+					.iter()
+					.map(|&pk| {
+						(
+							pk.into(),
+							SessionKeys { aura: sr25519::Public::from_raw(pk).into() }.into(),
+						)
+					})
+					.collect();
+
+				pallet_session::QueuedKeys::<T>::put(queued_keys);
+
+				for pk in raw_keys.clone() {
+					pallet_session::NextKeys::<T>::insert::<
+						<T as pallet_session::Config>::ValidatorId,
+						<T as pallet_session::Config>::Keys,
+					>(pk.into(), SessionKeys { aura: sr25519::Public::from_raw(pk).into() }.into());
+					pallet_session::KeyOwner::<T>::insert::<
+						_,
+						<T as pallet_session::Config>::ValidatorId,
+					>((AURA, pk.encode()), pk.into());
 				}
-				log::info!(target: TARGET, "bruteforcing from {:?} to 1", onchain_version);
-				StorageVersion::new(1).put::<Pallet<T>>();
-
-				T::DbWeight::get().reads_writes(1, 1)
-			}
-
-			#[cfg(feature = "try-runtime")]
-			fn post_upgrade(_state: sp_std::vec::Vec<u8>) -> Result<(), TryRuntimeError> {
-				ensure!(StorageVersion::get::<Pallet<T>>() == 1, "Must upgrade");
-				Ok(())
-			}
-		}
-	}
-}
-pub mod scheduler {
-	// this is necessary because migrations from v0 to v3 are no longer available in the scheduler
-	// pallet code and migrating is only possible from v3. The strategy here is to empty the agenda
-	// (has been empty since genesis)
-	use frame_support::traits::OnRuntimeUpgrade;
-	use frame_system::pallet_prelude::BlockNumberFor;
-	use pallet_scheduler::*;
-	use sp_std::vec::Vec;
-
-	#[cfg(feature = "try-runtime")]
-	use sp_runtime::TryRuntimeError;
 
-	/// The log target.
-	const TARGET: &'static str = "runtime::fix::scheduler::migration";
+				let invulnerables: Vec<<T as frame_system::Config>::AccountId> =
+					raw_keys.iter().map(|&pk| pk.into()).collect();
+				let invulnerables: BoundedVec<_, T::MaxInvulnerables> =
+					invulnerables.try_into().unwrap();
+				pallet_collator_selection::Invulnerables::<T>::put(invulnerables);
 
-	pub mod v1 {
-		use super::*;
-		use frame_support::{pallet_prelude::*, traits::schedule};
-
-		#[cfg_attr(any(feature = "std", test), derive(PartialEq, Eq))]
-		#[derive(Clone, RuntimeDebug, Encode, Decode)]
-		pub(crate) struct ScheduledV1<Call, BlockNumber> {
-			maybe_id: Option<Vec<u8>>,
-			priority: schedule::Priority,
-			call: Call,
-			maybe_periodic: Option<schedule::Period<BlockNumber>>,
-		}
+				pallet_collator_selection::CandidacyBond::<T>::put::<BalanceOf<T>>(
+					533_333_328u128.into(),
+				);
```

Comment:
IIRC it's 50 KSM for other system chains. Perhaps put the same value there?

Reply:
Ok. Seems I copy-pasted from the wrong place. Will check

Reply:
As we'd like individual Encointer communities to run collators, 50 KSM seems too much and will likely prevent self-sustained collators running in the global south. I set it to 5 KSM instead for starters.

Reply:
Since it uses an election mechanism now, we can probably reduce the other chains to 5 as well and just let the market find what it's worth to be a collator.